src/components/typeahead/typeahead.component.ts

```diff
@@ -174,6 +183,17 @@ export class TypeaheadComponent<T = any> implements OnChanges, OnDestroy {
                 }
             });
 
+        this._popoverOrientationDirection = this._popoverOrientation.createPopoverOrientationListener(this.typeaheadElement.nativeElement, this.typeaheadElement.nativeElement.parentElement);
+
+        this._popoverOrientationDirection.orientation$.pipe(takeUntil(this._onDestroy))
+            .subscribe(direction => {
+                if (this.dropDirection === 'auto' && direction === PopoverOrientationDirection.Up) {

```

Comment:
Just noticed that dropDirection = 'up' no longer works, I think this code will need to change. I'd suggest switching away from using renderer and instead computing a property with a hostbinding, e.g.:
```ts
@HostBinding('class.drop-up')
private _dropUp: boolean;
```

You can then update the property from both sources, e.g. set up ngOnChanges to watch `dropDirection`:
```ts
if (changes.dropDirection.currentValue === 'auto') {
  this._dropUp = this._popoverOrientationDirection.orientation$.getValue() === PopoverOrientationDirection.Up;
} else {
  this._dropUp = changes.dropDirection.currentValue === 'up';
}
```

This subscription should then only update `_dropUp` if `dropDirection` is auto.

Reply:
I tried doing the above but because I am using orientation$ is a Subject I don't have access to valueOf() and some of the times the current value was coming back as undefined. I've pushed a fix that has fixed the 'up' dropDirection.

Reply:
You can convert orientation$ to a BehaviorSubject to resolve that.